snuba/query/parser/functions.py

```diff
@@ -0,0 +1,105 @@
+import numbers
+from datetime import date, datetime
+from typing import Any, Callable, List, Optional, TypeVar, Tuple, Union
+
+from snuba.query.expressions import Column, Expression, Literal, FunctionCall
+from snuba.util import is_function, QUOTED_LITERAL_RE
+
+TExpression = TypeVar("TExpression")
+
+
+def parse_function(
+    output_builder: Callable[[Optional[str], str, List[TExpression]], TExpression],
+    simple_expression_builder: Callable[[str], TExpression],
+    literal_builder: Callable[
+        [Optional[Union[str, datetime, date, List[Any], Tuple[Any], numbers.Number]]],
```

Comment:
extreme nit: I don't think we are planning on supporting all types of number (e.g. `Complex`.)

Reply:
Indeeed, unfortunately that signature is required by this method 
https://github.com/getsentry/snuba/blob/master/snuba/util.py#L202-L222

which is called by that function